ape_solidity/compiler.py

```diff
@@ -738,6 +729,50 @@ def enrich_error(self, err: ContractLogicError) -> ContractLogicError:
                 txn=err.txn,
             )
 
+    def _flatten_source(self, path: Path, base_path=None) -> str:
+        base = base_path or path.parent
+        imports = self.get_imports_with_raw([path])
+        source = ""
+        for import_list in imports.values():
+            if not import_list:
+                continue
+            for import_path, raw_import_path in import_list:
+                source += f"// File: {raw_import_path}\n\n"
+                source += self._flatten_source(base / import_path, base_path=base)
+        if base_path is None:
+            source += f"\n// File: {path.name}\n\n"
+        source += path.read_text() + "\n"
+        return source
+
+    def flatten_contract(self, path: Path, **kwargs) -> Content:
+        """Flatten a contract.
+
+        Args:
+            path: Path to contract file.
+            **kwargs: Keyword arguments
+
+        Returns:
+            Content of flattened contract.
+        """
+        # try compiling in order to validate it works
+        self.compile([path], base_path=self.config_manager.contracts_folder)
+        source = self._flatten_source(path)
+        source = remove_imports(source)
+        lines = source.splitlines()
+        line_dict = {i + 1: line for i, line in enumerate(lines)}
+        return Content(__root__=line_dict)
+
+
+def remove_imports(flattened_contract: str) -> str:
+    # Define a regex pattern that matches import statements
+    # Both single and multi-line imports will be matched
+    pattern = r"import\s+((.*?)(?=;)|[\s\S]*?from\s+(.*?)(?=;));\s"
```

Comment:
as a small perf, you could do

```python
IMPORTS_PATTERN = re.compile(r"import\s+((.*?)(?=;)|[\s\S]*?from\s+(.*?)(?=;));\s")
```

at the top of the module and use it here like 

```python
IMPORTS_PATTERN.sub("", flattened_contract, flags=re.MULTILINE)
```